source/fuzz/repeated_pass_recommender_standard.cpp

```diff
@@ -0,0 +1,325 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/repeated_pass_recommender_standard.h"
+
+#include <numeric>
+
+namespace spvtools {
+namespace fuzz {
+
+RepeatedPassRecommenderStandard::RepeatedPassRecommenderStandard(
+    RepeatedPassInstances* pass_instances, FuzzerContext* fuzzer_context)
+    : pass_instances_(pass_instances), fuzzer_context_(fuzzer_context) {}
+
+RepeatedPassRecommenderStandard::~RepeatedPassRecommenderStandard() = default;
+
+std::vector<FuzzerPass*>
+RepeatedPassRecommenderStandard::GetFuturePassRecommendations(
+    const FuzzerPass& pass) {
+  if (&pass == pass_instances_->GetAddAccessChains()) {
+    // - Adding access chains means there is more scope for loading and storing
+    // - It could be worth making more access chains from the recently-added
+    //   access chains
+    return RandomOrderAndNonNull({pass_instances_->GetAddLoads(),
+                                  pass_instances_->GetAddStores(),
+                                  pass_instances_->GetAddAccessChains()});
+  }
+  if (&pass == pass_instances_->GetAddBitInstructionSynonyms()) {
+    // - Adding bit instruction synonyms creates opportunities to apply synonyms
+    return RandomOrderAndNonNull({pass_instances_->GetApplyIdSynonyms()});
+  }
+  if (&pass == pass_instances_->GetAddCompositeInserts()) {
+    // - Having added inserts we will have more vectors, so there is scope for
+    //   vector shuffling
+    // - Adding inserts creates synonyms, which we should try to use
+    // - Vector inserts can be made dynamic
+    return RandomOrderAndNonNull(
+        {pass_instances_->GetAddVectorShuffleInstructions(),
+         pass_instances_->GetApplyIdSynonyms(),
+         pass_instances_->GetMakeVectorOperationsDynamic()});
+  }
+  if (&pass == pass_instances_->GetAddCompositeTypes()) {
+    // - More composite types gives more scope for constructing composites
+    return RandomOrderAndNonNull({pass_instances_->GetConstructComposites()});
+  }
+  if (&pass == pass_instances_->GetAddCopyMemory()) {
+    // - Recently-added copy memories could be replace with load-store pairs
+    return RandomOrderAndNonNull(
+        {pass_instances_->GetReplaceCopyMemoriesWithLoadsStores()});
+  }
+  if (&pass == pass_instances_->GetAddDeadBlocks()) {
+    // - Dead blocks are great for adding function calls
+    // - Dead blocks are also great for adding loads and stores
+    // - The guard associated with a dead block can be obfuscated
+    return RandomOrderAndNonNull({pass_instances_->GetAddFunctionCalls(),
+                                  pass_instances_->GetAddLoads(),
+                                  pass_instances_->GetAddStores(),
+                                  pass_instances_->GetObfuscateConstants()});
```

Comment:
We could probably add a pass here that replaces uses of irrelevant ids since all ids in dead blocks are considered irrelevant.

Reply:
The pass that adds dead blocks doesn't immediately put anything in them, so there wouldn't be irrelevant ids to replace at this point.